ui/prometheus-plugin/src/plugins/range-chart-query.ts

```diff
@@ -54,13 +55,32 @@ export function usePrometheusRangeChartQuery(
   const timeRange = useDashboardPrometheusTimeRange();
   const step = usePanelRangeStep(timeRange, minStep);
 
+  // Align the time range so that it's a multiple of the step (TODO: we may
+  // ultimately want to return this from the hook so that charts will know what
+  // time range was actually used?)
```

Comment:
Yep! We'll eventually want that knowledge available to whatever paints the chart, so it can show exactly that range, no matter how much data is being returned :)